go/batch-submitter/drivers/sequencer/encoding_test.go

```diff
@@ -0,0 +1,305 @@
+				"c9808080808080808080",
+				"c9808080808080808080",
+				"c9808080808080808080",
+			},
+		},
+		{
+			Name: "single context",
+			HexEncoding: "0000000001000000" +
+				"000001" +
+				"000102030405060708090a0b0c0d0e0f",
+			ShouldStartAtElement:  1,
+			TotalElementsToAppend: 0,
+			Contexts: []sequencer.BatchContext{
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+			},
+			Txs: nil,
+		},
+		{
+			Name: "multiple contexts",
+			HexEncoding: "0000000001000000" +
+				"000004" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"000102030405060708090a0b0c0d0e0f",
+			ShouldStartAtElement:  1,
+			TotalElementsToAppend: 0,
+			Contexts: []sequencer.BatchContext{
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+			},
+			Txs: nil,
+		},
+		{
+			Name: "complex",
+			HexEncoding: "0102030405060708" +
+				"000004" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"000102030405060708090a0b0c0d0e0f" +
+				"00000ac9808080808080808080" +
+				"00000ac9808080808080808080" +
+				"00000ac9808080808080808080" +
+				"00000ac9808080808080808080",
+			ShouldStartAtElement:  0x0102030405,
+			TotalElementsToAppend: 0x060708,
+			Contexts: []sequencer.BatchContext{
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+				{
+					NumSequencedTxs:       0x000102,
+					NumSubsequentQueueTxs: 0x030405,
+					Timestamp:             0x060708090a,
+					BlockNumber:           0x0b0c0d0e0f,
+				},
+			},
+			Txs: []string{
+				"c9808080808080808080",
+				"c9808080808080808080",
+				"c9808080808080808080",
+				"c9808080808080808080",
```

Comment:
Confirming that this is an empty transaction:

```
$ rlp decode 0xc9808080808080808080
[
  '', '', '', '', '',
  '', '', '', ''
]
```

Reply:
yes, i just used the most minimal txn possible for the test vectors. we could add more complicated ones, plus invalid ones for negative test vectors.